main/src/main/scala/sbt/Defaults.scala

```diff
@@ -1338,7 +1341,7 @@ object Defaults extends BuildCommon {
   def singleTestGroup(key: Scoped): Initialize[Task[Seq[Tests.Group]]] =
     inTask(key, singleTestGroupDefault)
   def singleTestGroupDefault: Initialize[Task[Seq[Tests.Group]]] = Def.task {
-    val tests = definedTests.value
+    val tests = definedTests.value.allDefinitions
```

Comment:
and e.g. here information about frameworks is discarded. However later mapping `framework -> test` has to be computed, but I don't know if we could do better (without significant code repetition) since changing `Tests.Group` would break compatibility :/